server/sublist.go

```diff
@@ -83,7 +83,7 @@ func NewSublist() *Sublist {
 // Insert adds a subscription into the sublist
 func (s *Sublist) Insert(sub *subscription) error {
 	// copy the subject since we hold this and this might be part of a large byte slice.
-	subject := string(append([]byte(nil), sub.subject...))
+	subject := string(sub.subject)
```

Comment:
The current reason the compiler escapes the string is this:

```
1352 ./sublist.go:86: string(sub.subject) escapes to heap                                                                                                                                                          
1353 ./sublist.go:86:    from subject (assigned) at ./sublist.go:86
1354 ./sublist.go:86:    from append(tokens, subject[start:i]) (appended to slice) at ./sublist.go:92
```

I got this by using 'go build -gcflags="-m -m" .' (the double -m is new in Go 1.7, I used the current tip).

But even without this, the call to s.addToCache would force a string to be allocated.